### ISSUE TITLE
Core node restart caused 11 replica nodes to restart, resulting in 8 million connections being disconnected

### ISSUE THREAD

**Issue by Estheriii** [emqx/emqx#11310]
### What happened?

A cluster with 3 core nodes and 18 replica nodes has 13 million connections. Manually restarting a core node will cause 11 replica nodes to restart and 8 million connections to be disconnected

### What did you expect to happen?

One-third of the connections are disconnected, affecting up to 6 connections

### How can we reproduce it (as minimally and precisely as possible)?

start a cluster with operator-controller in k8s cluster
with 3 core and 18 replica
add 13 million connections
than delete a core node 

### EMQX version

<details>
5.0.21
```console
$ ./bin/emqx_ctl broker
# paste output here
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Log files

<details>

</details>
[log.zip](https://github.com/emqx/emqx/files/12101389/log.zip)

**Comment by id** [comment]
@Estheriii I do not think it's a bug. If you do manual restart of a core node without [node evacuation](https://docs.emqx.com/en/enterprise/v5.1/deploy/cluster/rebalancing.html#node-evacuation-and-cluster-load-rebalancing), it is expected that the connections to that node will be dropped.

I would recommend to discuss this issue on the [emqx-operator forum](https://github.com/emqx/emqx-operator/discussions/).
This issue may be related: https://github.com/emqx/emqx-operator/discussions/628

**Comment by Estheriii** [comment]
> @Estheriii I do not think it's a bug. If you do manual restart of a core node without [node evacuation](https://docs.emqx.com/en/enterprise/v5.1/deploy/cluster/rebalancing.html#node-evacuation-and-cluster-load-rebalancing), it is expected that the connections to that node will be dropped.
> 
> I would recommend to discuss this issue on the [emqx-operator forum](https://github.com/emqx/emqx-operator/discussions/). This issue may be related: [emqx/emqx-operator#628](https://github.com/emqx/emqx-operator/discussions/628)

I am simulating a node crash situation where a large number of connections are disconnected due to a node restart, indicating that such a cluster is not highly available

**Comment by id** [comment]
Were clients not able to reconnect after you restarted the core node, or did you loose some data?

**Comment by id** [comment]
> indicating that such a cluster is not highly available

When you manually restart a node, all open sockets on that machine will be forcibly closed. There is no way to maintain the connections when you bring the node down.
High availability means that the cluster as a whole remains operational and available, and clients are able to re-establish connection and continue working even when some nodes in the cluster are down.

**Comment by Estheriii** [comment]
> > indicating that such a cluster is not highly available
> 
> When you manually restart a node, all open sockets on that machine will be forcibly closed. There is no way to maintain the connections when you bring the node down. High availability means that the cluster as a whole remains operational and available, and clients are able to re-establish connection and continue working even when some nodes in the cluster are down.

yes so i can accept 6 nodes down with 6 replica nodes restart and 4million connections disconnected but not accept 11 replica nodes restart and 8 million connections disconnected

**Comment by Estheriii** [comment]
> > indicating that such a cluster is not highly available
> 
> When you manually restart a node, all open sockets on that machine will be forcibly closed. There is no way to maintain the connections when you bring the node down. High availability means that the cluster as a whole remains operational and available, and clients are able to re-establish connection and continue working even when some nodes in the cluster are down.

Oh! I just realized that restarting one core node should not cause any replica nodes to restart, as three core nodes should be highly available.

**Comment by richard-scott** [comment]
I've just spun up a 3-core, 6-replica cluster in AWS/EC2 and tested by rebooting (reboot -f to force a fast reboot) a Core node and all Replicas stay online with all clients still connected.

@Estheriii What is your infrastructure setup? How/where are you running these nodes?

**Comment by Estheriii** [comment]
> I've just spun up a 3-core, 6-replica cluster in AWS/EC2 and tested by rebooting (reboot -f to force a fast reboot) a Core node and all Replicas stay online with all clients still connected.
> 
> @Estheriii What is your infrastructure setup? How/where are you running these nodes?

use operator 2.1.1 to start a cluster in k8s , must have a million level connection, with the replica node occupying 70% of the CPU

**Comment by elysian-gc** [comment]
I've got 4 million connections by [emqtt_bench](https://github.com/emqx/emqtt-bench), then stop a core node, and all connections stay onlin. 
@Estheriii  Can you reproduce it stably?
![image](https://github.com/emqx/emqx/assets/44466800/c5535425-f2b4-4700-803e-09b0321aec91)

**Comment by elysian-gc** [comment]
I tried to restart core emqx , and the result is still

**Comment by SergeTupchiy** [comment]
Hi @Estheriii,
Thanks for reporting this issue, we have managed to reproduce it and are working on the fix.

**Comment by elysian-gc** [comment]
@SergeTupchiy  could you tell me how to reproduce it? and which version can fix, thanks

**Comment by brucejiao777** [comment]
Hope all is well and give you a hint when this issue is fixed

**Comment by id** [comment]
Should be fixed in the latest release v5.1.5-build.3 @Estheriii. Please check.
